backend/Application/Api/GraphQL/Import/ImportWriteController.cs

```diff
@@ -185,9 +185,25 @@ private async Task<BlockWriteResult> HandleCommonWrites(BlockDataPayload payload
 
         var passiveDelegationUpdateResults = await _passiveDelegationHandler.UpdatePassiveDelegation(delegationUpdateResults, payload, importState, importPaydayStatus, block);
         await _bakerHandler.ApplyChangesAfterBlocksAndTransactionsWritten(block, transactions, importPaydayStatus);
-        _accountHandler.HandleAccountUpdates(payload, transactions, block);
+        var accountBalanceUpdates = _accountHandler.HandleAccountUpdates(payload, transactions, block);
+        var accountTokenUpdates = _eventLogHandler.HandleLogs(transactions);
 
-        _eventLogHandler.HandleLogs(transactions);
+        var updatedAccountAddresses = accountTokenUpdates
+            .Select(u => u.Address)
+            .Concat(accountBalanceUpdates.Select(a => a.AccountAddress))
+            .Distinct()
+            .ToArray();
+
+        if (updatedAccountAddresses.Length > 0)
+        {
```

Comment:
why is this if necessary? Won't it just be an empty loop otherwise?

Reply:
removed, thanks